### ISSUE TITLE
Implement LavaMoat / freeze underlying packages

### ISSUE THREAD

**Issue by hstove** [leather-io/extension#644]
We would like to implement [LavaMoat](https://github.com/LavaMoat/lavamoat) in this repository to prevent supply chain attacks.

**Comment by psq** [comment]
I'm not so sure this should be included in libraries others can use, this should rather be used in the app itself, should the developer chose to do so, not be imposed by dependencies...  Unless it would not leak outside the library context, which I'm not sure is possible.

**Comment by hstove** [comment]
Yes, the idea is to have this only in place for the authenticator itself.

**Comment by markmhendrickson** [comment]
Related discussion from @beguene: https://github.com/blockstack/stacks-wallet-web/discussions/1347
